### PR TITLE
Bug1523744 macos logging

### PR DESCRIPTION
**Pull request by davehouse**
* collect logs from macos unified logging
* forward logs to stackdriver

1. installs fluentd (td-agent) from dmg in s3
2. installs stackdriver plugin (and required coreutils from homebrew)
3. configures to collect logs from apple's `log stream` commandline tool
4. configures forwarding logs to stackdriver
  4a. adds auth for stackdriver service account to write logs

### CODE REVIEW

File: modules/fluentd/templates/fluentd.conf.erb [hunk 1]
```diff
@@ -0,0 +1,84 @@
+<source>
+  @type exec
+  command "log stream --color none --level debug --type log --style syslog \
+      --timeout 1d \
+    | tail -n +3 \
+    | awk '{printf (/^[0-9]{4}/&&NR>1?RS:x)\"%s\",$0} END {print \"\"}'"
+    # tail to omit first 2 lines that cause syslog parsing to fail:
+    #   Filtering the log data using "type == 1024"
+    #   Timestamp                       (process)[PID]    
+    # awk remove newlines apple's log output as syslog prints in log messages
```
Comment (davehouse) [inline modules/fluentd/templates/fluentd.conf.erb]: To simplify this (removing the step with awk), we could output json and capture specific fields. However, Json input for fluentd expects to not have the entries in an array: some string manipulation will still be required.

File: modules/fluentd/manifests/init.pp [hunk 2]
```diff
@@ -0,0 +1,61 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+class fluentd (
+    String $worker_type,
+    String $stackdriver_project,
+) {
+
+    $stackdriver_keyid    = lookup("stackdriver.${stackdriver_project}.keyid")
```
Comment (davehouse) [inline modules/fluentd/manifests/init.pp]: I added these secrets to the vault.yaml on install2 mdc1/2

File: modules/fluentd/templates/fluentd.conf.erb (hunk 1 above)
Comment (davehouse) [reply modules/fluentd/templates/fluentd.conf.erb]: I switched to the default log style to get the log level/severity.

File: modules/fluentd/manifests/init.pp (hunk 2 above)
Comment (dividehex) [reply modules/fluentd/manifests/init.pp]: As we discussed over video, I'd like to see these lookups done from a profile and passed as parameters to the fluentd class.  A generic 'logging' profile should be used for all roles.

File: modules/roles_profiles/manifests/profiles/gecko_t_osx_1014_generic_worker.pp [hunk 3]
```diff
@@ -6,11 +6,18 @@
...
     require roles_profiles::profiles::cltbld_user
 
+    $worker_type = 'gecko-t-osx-1014'
+
     case $::operatingsystem {
         'Darwin': {
 
             include puppet::atboot
 
+            class { 'fluentd':
```
Comment (dividehex) [inline modules/roles_profiles/manifests/profiles/gecko_t_osx_1014_generic_worker.pp]: Move this to a generic 'logging' profile and then include it on the gecko 1014 role.

**Comment by davehouse** [comment]
t-mojave-r7-467.test.releng.mdc1.mozilla.com is running with this config

**Comment by davehouse** [comment]
Thank you! I'll reference your note for a revision/update

**Pull request merged**